cartographer_ros/cartographer_ros/offline_node.cc

```diff
@@ -309,9 +313,21 @@ void RunOfflineNode(const MapBuilderFactory& map_builder_factory) {
 #endif
 
   if (::ros::ok()) {
-    const std::string output_filename = bag_filenames.front() + ".pbstream";
-    LOG(INFO) << "Writing state to '" << output_filename << "'...";
-    node.SerializeState(output_filename);
+    std::string output_filename;
+    const std::string suffix = ".pbstream";
+    if (bag_filenames.size() > 0) {
+      output_filename = bag_filenames.front();
+    } else {
```

Comment:
It seems to me that when you load N .pbstreams in unfrozen state, they are not optimized relative to one another (no inter pbstream constraints are searched for). Given that why would anyone want to write a pbstream in a situation where no bag file has been played?

Reply:
It's useful to be able to launch an offline node instance which only serves a loaded unfrozen `.pbstream` for visualization (using `keep_running = true`). However, you're right about further serialization making no sense in that scenario. I'll change it to avoid serializing when no bags are played.

Reply:
OK, SG. Having N .pbstream of the same are and being able to fire up offline node to optimize them with respect to one another might be useful though. So maybe we add that in the future. But for now agreed remove.

Reply:
Sounds like an interesting idea. Maybe it belongs in an RFC :)